### DISCUSSION TITLE
Why my hocuspocus server can only accept connection by ip address instead of domain name?

### DISCUSSION THREAD

**Discussion by lanyusan** [ueberdosis/hocuspocus#619] [Q&A] [answered]
I have a standard hocuspocus server:
```
import { Redis } from "@hocuspocus/extension-redis";
import { Hocuspocus } from "@hocuspocus/server";
import { createClient, commandOptions } from "redis";
import { Database } from "@hocuspocus/extension-database";

const client = createClient();

client.on("error", (err) => console.log("Redis Client Error", err));

await client.connect();

// Configure the server …
const server = new Hocuspocus({
  port: 7654,

  extensions: [
    new Database({
      // Return a Promise to retrieve data …
      fetch: async ({ documentName }) => {
        const value = await client.get(
          commandOptions({ returnBuffers: true }),
          documentName
        );
        // if (value) return new Uint8Array(value.buffer);
        // else return null;
        return value;
      },
      // … and a Promise to store data:
      store: async ({ documentName, state }) => {
        await client.set(documentName, state);
      },
    }),

    new Redis({
      host: "127.0.0.1", // make sure to use the same Redis instance :-)
      port: 6379,
    }),
  ],

  async onAuthenticate({ documentName, token }) {
    // Could be an API call, DB query or whatever …
    // The endpoint should return 200 OK in case the user is authenticated, and an http error
    // in case the user is not.
    // return axios.get("/user", {
    //   headers: {
    //     Authorization: `Bearer ${token}`,
    //   },
    // });

    return {
      user: {
        id: 1234,
        name: "John",
      },
    };
  },
});

// … and run it!
server.listen();

```

which works fine when client is connecting by `ws://localhost:7654`.

When I deploy it to server, remote client can connect by `ws://123.123.123.123:7654` just fine.

When I try to connect by domain name `ws://abc.com:7654`, then connection always fail. 

Why is that happening? I am pretty sure that the domain name has been bound to server address. I have tested the connection with telnet, which works just fine.

Can someone please help?

Thanks.

**Comment by mkriegeskorte** [comment]
Hard to tell without further insights into your setup.

>  am pretty sure that the domain name has been bound to server address.

Make sure this is really the case, otherwise it cannot work.

Do you use any kind of proxy like nginx in front of the server or other services?

Maybe the browser prevents the request as you’re trying to connect to a insecure websocket address. (Check your network tab)

However, it should definitely work with a domain name.

**Comment by lanyusan** [comment, marked as answer]
Thanks for tip.
It was indeed a network setup problem at our server side.

**Reply by DineshCodeFlow** [reply]
The same problem is with us. What network issue are you talking about

**Comment by DineshCodeFlow** [comment]
I am reverse proxying it using caddy. It’s simple.
